lib/dal/src/cached_module.rs

```diff
@@ -561,143 +522,199 @@ impl CachedModule {
             .await?
             .pg()
             .query_one(
-                query,
+                &query,
                 &[
                     &schema_id,
-                    &schema_name,
-                    &display_name,
-                    &category,
-                    &link,
-                    &color,
-                    &description,
-                    &component_type.to_string(),
+                    &package
+                        .schema_name()
+                        .unwrap_or(module_details.name.as_str()),
+                    &package.display_name(),
+                    &package.category(),
+                    &package.link(),
+                    &package.color(),
+                    &package.description(),
+                    &package.component_type().to_string(),
+                    &package.package_summary(),
                     &module_details.latest_hash,
                     &module_details.created_at,
                     &bytes_ref,
-                    &user_pk,
+                    &scoped_to_user_pk,
                 ],
             )
             .await?;
 
         Ok(Some(row.try_into()?))
     }
 
-    async fn insert(
+    pub async fn remove_all_for_schema_id(
         ctx: &DalContext,
-        module_details: &ModuleDetailsResponse,
-        pkg_bytes: Arc<Vec<u8>>,
-    ) -> CachedModuleResult<Option<Self>> {
-        let bytes_clone = pkg_bytes.clone();
-        let pkg = slow_rt::spawn(async move { SiPkg::load_from_bytes(&bytes_clone) })?.await??;
+        schema_id: SchemaId,
+    ) -> CachedModuleResult<()> {
+        let query = r#"
+            DELETE FROM cached_modules WHERE schema_id = $1
+            "#;
+
+        ctx.txns().await?.pg().query(query, &[&schema_id]).await?;
+
+        Ok(())
+    }
 
+    // module_id is just for debugging purposes; we only want one record per hash
+    pub async fn update_missing_package_summaries(ctx: &DalContext) -> CachedModuleResult<()> {
```

Comment:
This goes through packages that are missing `package_summary`, and fills them in. Used for sdf migrate, but one imagines the loop will be used later to detect and correct incorrect package versions.